src/api/foreground/foreground-setup-wallet-sdk.ts

```diff
@@ -177,4 +167,66 @@ export function setupWalletSDK(
       events.emit(e.data.event.name, e.data.event.value);
     }
   );
+
+  // at the end of the injected script,
+  // we dispatch the wallet loaded event
+
+  async function dispatchArweaveWalletLoaded() {
+    if (!window.arweaveWallet) return;
+
+    const permissions = await window.arweaveWallet
+      .getPermissions()
+      .catch(() => []);
+
+    // Note that for Wander Connect we just need to dispatch this once, no need to subscribe to the window load event to re-dispatch it:
+    dispatchEvent(
+      new CustomEvent("arweaveWalletLoaded", {
+        detail: {
+          permissions
+        }
+      })
+    );
+
+    if (permissions.length > 0) {
```

Comment:
@7i7o I think it makes sense that when a dApp is connected, it receives the `arweaveWalletLoaded` as well as all the other events, as that gives it the current state of the wallet.

However, with the current implementation, if I add a new listener to `arweaveWalletLoaded` way later after the page has loaded, all these other events are dispatched again. Maybe instead we could just include all this in the `arweaveWalletLoaded` details (permissions, activeAddress, addresses, gateway, ...).

Reply:
https://communitylabs.atlassian.net/browse/ARC-1276